src/middlewared/middlewared/plugins/tunables.py

```diff
@@ -25,212 +26,110 @@ class TunableService(CRUDService):
     class Config:
         datastore = 'system.tunable'
         datastore_prefix = 'tun_'
-        datastore_extend = 'tunable.upper'
         cli_namespace = 'system.tunable'
 
-    def __init__(self, *args, **kwargs):
-        super(TunableService, self).__init__(*args, **kwargs)
-        self.__default_sysctl = {}
+    SYSTEM_TUNABLES = deque()
 
     ENTRY = Patch(
         'tunable_create', 'tunable_entry',
         ('add', Int('id')),
     )
 
     @private
-    async def default_sysctl_config(self):
-        return self.__default_sysctl
+    def get_system_tunables(self):
+        if not TunableService.SYSTEM_TUNABLES:
+            tunables = subprocess.run(['sysctl', '-aN'], stdout=subprocess.PIPE)
+            for tunable in filter(lambda x: x, tunables.stdout.decode().split('\n')):
+                TunableService.SYSTEM_TUNABLES.append(tunable)
+        return TunableService.SYSTEM_TUNABLES
 
     @private
-    async def get_default_value(self, oid):
-        return self.__default_sysctl[oid]
+    def get_original_value(self, var):
+        with open(f'/proc/sys/{var.replace(".", "/")}') as f:
```

Comment:
Do we have a guarantee that this file will always exist ? And if not, can we add some exception handling please ?

Reply:
The reason why I didn't is that this is a private method and if you look where it's being called from then we make sure that the tunable actually exist in `SYSTEM_TUNABLES`. This means the file will exist any other exception and it should crash here.

The only exception to this is if someone upgrades from core to scale and they have a bunch of sysctls set then if they try and `do_update` those it will crash. However, that will happen with or without these changes. I will be pushing another PR to try and fix the core to scale upgrade situation but wanted to get this done first.